scala/scala.bzl

```diff
@@ -257,56 +253,53 @@ scala_repl = rule(
     toolchains = ['@io_bazel_rules_scala//scala:toolchain_type'],
 )
 
-_SCALA_BUILD_FILE = """
-# scala.BUILD
-java_import(
-    name = "scala-xml",
-    jars = ["lib/scala-xml_2.11-1.0.5.jar"],
-    visibility = ["//visibility:public"],
-)
+def scala_repositories(scala_version = "2.11.11"):
+  major_version_underscore = scala_version[:scala_version.find(".", 2)].replace(
+      ".", "_")
 
-java_import(
-    name = "scala-parser-combinators",
-    jars = ["lib/scala-parser-combinators_2.11-1.0.4.jar"],
-    visibility = ["//visibility:public"],
-)
-
-java_import(
-    name = "scala-library",
-    jars = ["lib/scala-library.jar"],
-    visibility = ["//visibility:public"],
-)
+  _new_scala_repository("scala_default", scala_version)
 
-java_import(
-    name = "scala-compiler",
-    jars = ["lib/scala-compiler.jar"],
-    visibility = ["//visibility:public"],
-)
-
-java_import(
-    name = "scala-reflect",
-    jars = ["lib/scala-reflect.jar"],
-    visibility = ["//visibility:public"],
-)
-"""
-
-def scala_repositories():
-  native.new_http_archive(
-      name = "scala",
-      strip_prefix = "scala-2.11.11",
-      sha256 =
-      "12037ca64c68468e717e950f47fc77d5ceae5e74e3bdca56f6d02fd5bfd6900b",
-      url = "https://downloads.lightbend.com/scala/2.11.11/scala-2.11.11.tgz",
-      build_file_content = _SCALA_BUILD_FILE,
+  # scalatest has macros, note http_jar is invoking ijar
+  native.http_jar(
+      name = "scalatest_2_11",
+      url =
+      "http://central.maven.org/maven2/org/scalatest/scalatest_2.11/3.0.5/scalatest_2.11-3.0.5.jar"
+  )
+  native.http_jar(
+      name = "scalactic_2_11",
+      url =
+      "http://central.maven.org/maven2/org/scalactic/scalactic_2.11/3.0.5/scalactic_2.11-3.0.5.jar"
   )
 
-  # scalatest has macros, note http_jar is invoking ijar
   native.http_jar(
-      name = "scalatest",
+      name = "scalatest_2_12",
       url =
-      "https://mirror.bazel.build/oss.sonatype.org/content/groups/public/org/scalatest/scalatest_2.11/2.2.6/scalatest_2.11-2.2.6.jar",
-      sha256 =
```

Comment:
let's make sure we find a solution to lock these back down with shas.

Maybe a dict for `Map[Version, Map[Name, Sha]]` something like that.

Reply:
In this case it's possible to just add the shas, just haven't done that. As we only need one scalatest version for 2.11 and one for 2.12 we can hardcode it, could ofc also be done programmatically with something like that

Reply:
Didn't actually think about 2.13. Makes sense to use a dict, to make it easier to add versions / allow the user to add a version without requiring changes in rules_scala